pandas/core/internals/construction.py

```diff
@@ -242,7 +242,10 @@ def init_dict(data, index, columns, dtype=None):
 
         # no obvious "empty" int column
         if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
+            if is_dtype_equal(dtype, "string"):
+                # GH 33623
+                nan_dtype = dtype
```

Comment:
this will be `dtype.na_value`

Reply:
can you update this

Reply:
Sorry, I can't figure out how to fix this from "this will be dtype.na_value".

Reply:
will change to `nan_dtype = dtype.na_value`, error occurs.

```
            if not isinstance(dtype, (np.dtype, type(np.dtype))):
>               dtype = dtype.dtype
E               AttributeError: 'NAType' object has no attribute 'dtype'
```

So I updated it like this.
```
if (
            dtype is None
            or is_extension_array_dtype(dtype)
            or np.issubdtype(dtype, np.flexible)
        ):
    nan_dtype = object
```